packages/common/src/types.ts

```diff
@@ -3,51 +3,25 @@ export interface Size {
   width: number;
 }
 
-export type CreatePluginEntity = (data: any) => object;
+export type EditorRef = any; // TODO
+export type ToolbarConfig = any; // TODO
 
-export interface EditorPluginProps<Insert> {
-  createEntityWithAsyncUploads?: CreateEntityWithAsyncUploads;
-  toolbarIcon: number;
-  getToolbarConfig: ({
-    entity,
-    updateEntity,
-    deleteEntity,
-    primaryColor,
-    entityType,
-    isPluginSpoilerAlertEnabled,
-  }: {
-    entity: any;
-    updateEntity: (data: any) => void;
-    deleteEntity: () => void;
-    primaryColor: string;
-    entityType: string;
-    isPluginSpoilerAlertEnabled: boolean;
-  }) => any;
-  ctaText: string;
-  config?: any;
-  onPress: (insert: Insert, insertAsync?: InsertAsync) => void;
-}
+export type CreatePluginEntity<RequiredEntityData, Entity> = (data: RequiredEntityData) => Entity;
 
-export interface DefaultEditorPluginProps {
-  pluginId: string;
+export interface EditorPluginProps {
+  toolbarIcon: number;
+  getToolbarConfig: (editorRef: EditorRef) => ToolbarConfig;
+  name: string;
+  onInsert: (editorRef: EditorRef) => void;
+  onInsertAsync: (editorRef: EditorRef) => void;
 }
 
-export type AtomicPlugin = {
+export type AtomicPlugin<RequiredEntityData, Entity> = {
   id: string;
-  ctaText: string;
-  pluginId: string;
+  name: string;
```

Comment:
I think maybe we should rename to `displayName` so it will be more clear this is for UI and not something else.